src/python/datastore/data_types.py

```diff
@@ -942,8 +945,13 @@ def get_environment_string(self):
 
     return environment_string
 
+  def populate_indices(self):
+    """Populate keywords for fast test case list searching."""
+    self.keywords = list(search_tokenizer.tokenize(self.name))
+
   def _pre_put_hook(self):
     """Pre-put hook."""
+    self.populate_indices()
     self.project = self.get_environment().get('PROJECT_NAME',
```

Comment:
actually, let's make it so that we can search by project as well.

Reply:
sure, will add that too.

Reply:
should we tokenize project name as well? It will be useful in cases such as `abseil-cpp` which will be tokenized to `abseil` and `cpp`

Reply:
Yes, good idea :)